src/pkg/utils/utils.go

```diff
@@ -183,3 +185,15 @@ func IsRegistryURL(s string) bool {
 
 	return false
 }
+
+// ReadYAMLStrict reads a YAML file into a struct, with strict parsing
+func ReadYAMLStrict(path string, destConfig any) error {
+	message.Debugf("Reading YAML at %s", path)
+
+	file, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	return goyaml.UnmarshalWithOptions(file, destConfig, goyaml.Strict())
```

Comment:
Nice and simple! I've tried this with a couple of mistyped bundles and it works as expected. Noting that we could get fancier and use [validators](https://github.com/defenseunicorns/zarf/blob/29d3ee39dfcbe173405f6e0010cd0dfb9468cd6e/src/pkg/packager/lint/lint.go#L269) like how Zarf does it, but this solution is really simple and it works